Add keyclock_file property to StreamManager

dz0080 opened the pull request:
Add keycloak_file to StreamerManager for Pravega instances requiring Keycloak authentication. The keycloak_file parameter can be passed to the create_stream_manager() function.

**Test**
build success, currently we don't have unit tests for c binding

dz0080 committed:
add keyclock support to StreamManager

vangork commented on c/src/stream_manager.rs:
```diff
@@ -21,23 +22,30 @@ impl StreamManager {
 
-        builder
+        let (auth_enabled, credential) = {
+            if keycloak_file.is_empty() {
```
if keycloack file is empty, don't set the credentials and let ClientConfigBuild to read from the env variables

vangork commented on c/src/stream_manager.rs:
```diff
@@ -124,16 +132,22 @@ pub unsafe extern "C" fn create_stream_manager_with_config(
-    let controller_uri = raw.to_str().unwrap();
+    let controller_uri = {
+        let cstr = CStr::from_ptr(controller_uri);
+        cstr.to_str().unwrap()
```
catch and handle unwrap

dz0080 replied:
Instead of panic or catch_unwind, currently return early on Err. This allow us to easily log individual errors if needed in the future.

vangork commented on c/src/stream_manager.rs:
```diff
@@ -124,16 +132,22 @@ pub unsafe extern "C" fn create_stream_manager_with_config(
+    };
+    let keycloak_file = {
+        let cstr = CStr::from_ptr(keycloak_file);
+        cstr.to_str().unwrap()
```
catch and handle unwrap

vangork commented on c/src/stream_manager.rs:
```diff
@@ -21,23 +22,30 @@ impl StreamManager {
         controller_uri: &str,
-        auth_enabled: bool,
+        keycloak_file: &str,
         tls_enabled: bool,
```
may remove tls_enabled.

dz0080 committed:
Add `create_stream_manager_with_keyclock`

This approach introduces `new_with_keycloak`, potentially leading to code duplication. Ideally, custom configurations for StreamManager should be created using `create_stream_manager_with_config`. However, the current implementation of `create_stream_manager_with_config` doesn't support optional values in `client_config`, making it difficult to set specific properties while using default values for others. A potential solution is refactoring `create_stream_manager_with_config` to support options.
